### PR TITLE
Create Facade Device for Shopper

### LINKED ISSUE #1: Using the provided materials create Shopper device class based on the facadedevice library.
Properties:
- [x] PlcAttrName_InterlockA
- [x] PlcAttrName_OpenS
- [x] PlcAttrName_ClosedS
- [x] PlcAttrName_CloseC
- [x] PlcAttrName_OpenC

Attributes:
- [x] ShopperOpen: PlcAttrName_OpenS readout.
- [x] ShopperClosed: PlcAttrName_ClosedS readout.
- [x] ShopperInterlock: PlcAttrName_InterlockA readout.

Commands:
- [x] Open: write True to PlcAttrName_OpenC.
- [x] Close: write True to PlcAttrName_CloseC.

### PR DESCRIPTION
**Pull request by stanislaw47**
closes #1 
* added project structure with scripts
* added documentation
* added `LICENSE` and updated `README`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,48 @@
...
+```
+
+How to run
+----------
+After installation, there is only one script: `Shopper`.
+You can use it via:
+```console
+Shopper instance_name
+```
+Remember that Device Server instance has to registered in database previously.
```
Comment (LJBD) [inline README.md]: Missing word: "(...) has to **be** registered (...)"

File: shopper_ds/shopper.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+# Imports
+from tango import DevState, AttrWriteType, DispLevel
+from facadedevice import Facade, proxy_attribute, proxy_command
+
+
+class Shopper(Facade):
+    """
+    This class implements Tango device server for control of shutter device (integrated shutter and stopper).
+    Each Tango device represents one shutter, which can be in open or closed.
```
Comment (LJBD) [inline shopper_ds/shopper.py]: Unnecessary word: "in"

File: shopper_ds/shopper.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    The Tango device works on a set of four PLC attributes of type DevShort, which must be
```
Comment (LJBD) [inline shopper_ds/shopper.py]: AFAIK those should be boolean attributes, not short. @wkitka can you confirm this?

File: shopper_ds/shopper.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                    "state.")
+
+    ShopperInterlock = proxy_attribute(
+        dtype=bool,
+        access=AttrWriteType.READ,
+        property_name="PLCAttrName_InterlockA",
+        display_level=DispLevel.OPERATOR,
+        description="Attribute that represents PLC signal for shutter in "
+                    "interlock alarm.")
+
```
Comment (LJBD) [inline shopper_ds/shopper.py:59]: Please add a method that will bind state & status to the interlock. When the interlock is on, the state should be `ALARM`.

File: shopper_ds/shopper.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    @proxy_command(
+        dtype_out=bool,
+        write_attribute=True,
+        property_name="PLCAttrName_OpenC",
+        doc_out="True to PLCAttrName_OpenC")
+    def Open(self, subcommand):
+        """
+         :rtype: bool
+        """
+        subcommand(1)
```
Comment (LJBD) [inline shopper_ds/shopper.py]: If you change type of the PLC attributes to boolean, you should change `1` to `True` here and in the `Close` command.

**Review by LJBD — CHANGES_REQUESTED** [review]

File: shopper_ds/shopper.py [hunk 6]
```diff
@@ -57,6 +57,15 @@ def safe_init_device(self):
         description="Attribute that represents PLC signal for shutter in "
                     "interlock alarm.")
 
+    # state attributes
+
+    @state_attribute(
+        bind=['ShopperInterlock'])
+    def InterlockAlarm(self, alarm):
```
Comment (LJBD) [inline shopper_ds/shopper.py]: Since it's a normal method and not a part of Tango interface, this method should be named using `snake_case`.

**Review by LJBD — CHANGES_REQUESTED** [review]

File: shopper_ds/shopper.py [hunk 7]
```diff
@@ -62,6 +62,13 @@ def safe_init_device(self):
     @state_attribute(
         bind=['ShopperInterlock'])
     def InterlockAlarm(self, alarm):
+        """
+        This attribute changes state of device, accordingly to ShopperInterlock
```
Comment (LJBD) [inline shopper_ds/shopper.py]: Technically, this isn't a Tango attribute, only the decorator contains `attribute` in its name 😄

Comment (stanislaw47) [reply shopper_ds/shopper.py]: So, `method` should be better word?

Comment (LJBD) [reply shopper_ds/shopper.py]: Definitely.

**Review by LJBD — APPROVED** [review]

**Pull request merged**